File: frontend-beheer-2024/src/types/flow.ts
import { FastApiResponse } from '.'
import { User } from './user'

export interface FlowRole {
  key: string
  alias: string
  min_required: number
  members: User[]
}

export interface Flow {
  key: string
  alias: string
  roles: FlowRole[]
}

export interface FlowResponse extends FastApiResponse {
  data: Flow
}

File: backend/app/schemas/flow.py
from pydantic import BaseModel

from app.middleware.authorisation.schemas import Role
from .user import User
from app.middleware.authorisation.config._base import Flow


class FlowStructureRole(BaseModel):
    key: Role
    alias: str
    min_required: int
    members: list[User]


class FlowStructure(BaseModel):
    key: Flow
    alias: str
    roles: list[FlowStructureRole]
